built_value_generator/lib/src/serializer_source_class.dart

```diff
@@ -29,13 +31,21 @@ abstract class SerializerSourceClass
 
   // TODO(davidmorgan): share common code in a nicer way.
   @memoized
-  BuiltValue get settings => new ValueSourceClass(element).settings;
+  BuiltValue get builtValueSettings => new ValueSourceClass(element).settings;
+
+  // TODO(davidmorgan): share common code in a nicer way.
+  @memoized
+  BuiltValueEnum get enumClassSettings => new EnumSourceClass(element).settings;
 
   @memoized
   String get name => element.name;
 
   @memoized
-  String get wireName => settings.wireName ?? name;
+  String get wireName =>
```

Comment:
If you use a { if... } it would be slightly more readable.

Reply:
Yup. Done!